firefox-ios/firefox-ios-tests/Tests/ClientTests/Frontend/Home/TopSites/UnifiedAdsCallbackTelemetryTests.swift

```diff
@@ -39,10 +43,47 @@ class UnifiedAdsCallbackTelemetryTests: XCTestCase {
         XCTAssertEqual(logger.savedMessage, "The unified ads telemetry call failed: \(tile.clickURL)")
     }
 
+    func testLegacyImpressionTelemetry() {
+        let subject = createSubject()
+        subject.sendImpressionTelemetry(tile: tile, position: 1)
+
+        XCTAssertEqual(gleanWrapper.recordQuantityCalled, 0)
+        XCTAssertEqual(gleanWrapper.recordStringCalled, 1)
+        XCTAssertEqual(gleanWrapper.recordUrlCalled, 0)
+        XCTAssertEqual(gleanWrapper.recordEventCalled, 1)
+        XCTAssertEqual(gleanWrapper.submitPingCalled, 1)
+        guard let savedPing = gleanWrapper.savedPing as? Ping<NoReasonCodes> else {
+            XCTFail("savedPing is not of type Ping<NoReasonCodes>")
+            return
+        }
+        XCTAssertEqual(asAnyHashable(savedPing), asAnyHashable(GleanMetrics.Pings.shared.topsitesImpression))
+        XCTAssertEqual(gleanWrapper.savedEvents?.count, 2)
```

Comment:
nit: to make this test more valuable, what about checking the array of event types instead of the count? Had to go back to prod code to confirm what the values should be for the count of 2.